### ISSUE TITLE
DATA BEGIN OF - faulty logic?

### ISSUE THREAD

**Issue by g-back** [abaplint/abaplint#680]
currently. the following does not give any errors:
```abap
DATA: BEGIN OF tab_ekpo,
        ebeln TYPE ekpo-ebeln,
        ebelp TYPE ekpo-ebelp,
      END OF tab_ekpo.
ebeln = 1.
ebelp = 2.
```
Reason: the components are seen as data definitions:

![image](https://user-images.githubusercontent.com/27279305/70713980-7f7ad180-1ce7-11ea-9f17-85acc0b4e186.png)

**Comment by larshp** [comment]
yea, looks like a bug

**Comment by larshp** [comment]
example reduced to

```abap
DATA: BEGIN OF foobar,
        moo TYPE i,
      END OF foobar.
moo = 1.
```

**Comment by larshp** [comment]
first part done, but needs more refactoring,

![image](https://user-images.githubusercontent.com/5888506/71319035-f27b0980-2498-11ea-910a-b067b4da4ea8.png)

**Comment by larshp** [comment]
similar problem with constants,

```abap
CONSTANTS: BEGIN OF bar,
             foo TYPE c LENGTH 1 VALUE 'a',
           END OF bar.
```
